src/daemon.rs

```diff
@@ -101,8 +123,14 @@ pub enum NotifyState {
     Errno(u8),
     /// A name for the submitted file descriptors.
     Fdname(String),
-    /// Stores additional file descriptors in the service manager.
+    /// Stores additional file descriptors in the service manager. Use [`notify_with_fds`] with this.
```

Comment:
Question on API design: should we maybe change this variant to be a `Fdstore(Vec<RawFd>)` or similar?

That way it sounds less likely to be misused (e.g. forgetting to set Fdstore or forgetting to pass the FDs), and at the same time it won't need a new dedicated `notify_with_fds()` function.

Reply:
That was my first attempt. It works but I didn't really like it because there are a lot of rules to notify. I felt it made the interface deceitful. You think you can combine multiple NotifyState in any way but that's far from the case. There are rules like these:

* A FDSTORE may send and number of file descriptors and be used with FDPOLL and FDNAME
* FDNAME and FDPOLL can be used once per call and is set for all filedescriptors.
* A BARRIER must be sent together with exactly one FD and nothing else.
* A FDSTOREREMOVE must be used with FDNAME.

What happens if multiple FDSTORE, or an FDSTORE and an FDSTOREREMOVE, is sent? Or a barrier? Then there's the "X_" class of notifications which I don't quite understand. But they could be sent with filedescriptors.

I think it would be much easier for the user of the library if there were functions such as:
```rust
fn notify_fdstore(fds: &[RawFd], name: Option<String>, automatic_cleanup: bool)
fn notify_fdremove(name: String)
fn notify_barrier(fd: RawFd)
fn notify_ready()
fn notify_reloading()
fn notify_status(s: String)
// etc
```

This removes the burden of understanding notify from the library user. Currently you have to read the sd_notify man page quite carefully to do things right.